kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx

```diff
@@ -94,22 +98,61 @@ export function EditReportDefinition(props) {
       });
   };
 
+  const editReportDefinition = async (metadata) => {
+    const { httpClient } = props;
+    /*
+      we check if this edit update the trigger type from Schedule to On demand. 
+      If so, need to first delete the curReportDefinition along with the scheduled job first, by calling the delete
+      report definition API
+    */
+    const {
+      trigger: { trigger_type: triggerType },
+    } = curReportDefinition;
+    if (
+      triggerType !== 'On demand' &&
+      metadata.trigger.trigger_type === 'On demand'
+    ) {
+      httpClient
+        .delete(`../api/reporting/reportDefinitions/${reportDefinitionId}`)
+        .then(async () => {
+          await callUpdateAPI(metadata);
+        })
+        .catch((error) => {
+          console.log('error when deleting report definition:', error);
```

Comment:
np: add a toast for when this error is triggered

Reply:
will add